core/eolearn/core/utilities.py

```diff
@@ -48,43 +48,60 @@ class FeatureParser:
 
     General guidelines:
-        - Almost all `EOTask`s have take as a parameter some information about features. The purpose of this class is
-        to unite and generalize parsing of such parameter over entire eo-learn package
-        - The idea for this class is that it should support more or less any logical way how to describe a collection
-        of features.
-        - Parameter `...` is used as a contextual clue. In the supported formats it is used to describe the most obvious
-        way how to specify certain parts of feature collection.
-        - Supports formats defined with lists, tuples, sets and dictionaries.
+
+    - Almost all EOTasks have take as a parameter some information about features. The purpose of this class is
+      to unite and generalize parsing of such parameter over entire eo-learn package
+    - The idea for this class is that it should support more or less any logical way how to describe a collection
+      of features.
+    - Parameter `...` is used as a contextual clue. In the supported formats it is used to describe the most obvious
+      way how to specify certain parts of feature collection.
+    - Supports formats defined with lists, tuples, sets and dictionaries.
 
     Supported input formats:
-        - `...` - Anything that exists in a given `EOPatch`
-        - A feature type describing all features of that type. E.g. `FeatureType.DATA` or `FeatureType.BBOX`
-        - A single feature as a tuple. E.g. (FeatureType.DATA, 'BANDS')
-        - A single feature as a tuple with new name. E.g. (FeatureType.DATA, 'BANDS', 'NEW_BANDS')
-        - A list of features (new names or not).
-        E.g. [(FeatureType.DATA, 'BANDS'), (FeatureType.MASK, 'CLOUD_MASK', 'NEW_CLOUD_MASK')]
-        - A dictionary with feature types as keys and lists, sets, single feature or `...` of feature names as values.
-        E.g. {
-            FeatureType.DATA: ['S2-BANDS', 'L8-BANDS'],
-            FeatureType.MASK: {'IS_VALID', 'IS_DATA'},
-            FeatureType.MASK_TIMELESS: 'LULC',
-            FeatureType.TIMESTAMP: ...
-        }
-        - A dictionary with feature types as keys and dictionaries, where feature names are mapped into new names, as
-          values.
-        E.g. {
-            FeatureType.DATA: {
-                'S2-BANDS': 'INTERPOLATED_S2_BANDS',
-                'L8-BANDS': 'INTERPOLATED_L8_BANDS',
-                'NDVI': ...
-            },
-        }
+
+    - Anything that exists in a given `EOPatch` is defined with `...`
+    - A feature type describing all features of that type. Example: `FeatureType.DATA` or `FeatureType.BBOX`
+    - A single feature as a tuple. Example: `(FeatureType.DATA, 'BANDS')`
+    - A single feature as a tuple. Example: `(FeatureType.DATA, 'BANDS')`
+    - A single feature as a tuple with new name. Example `(FeatureType.DATA, 'BANDS', 'NEW_BANDS')`
```

Comment:
🤔 Is this even true? 
If `new_names=False` such notation will result in two features of type `FeatureType.DATA` type, one named `'BANDS'` and another named `'NEW_BANDS'`. 

`(FeatureType.DATA: {'BANDS': 'NEW_BANDS'})` is much more clear notation, and possibly removes the need for `new_names` parameter altogether. The problem arises as the `FeatureParser` is typically used within an eo-learn task, and the user doesn't know if the `new_names` is used or not.

Reply:
Yes, the case where you have `(ft, fn1, fn2)` is the problematic case that can be interpreted in 2 different ways depending on parameter `new_names`. It should be avoided when trying to define multiple features.

We noticed this problem a few years back but never had the time to fix it. Probably a good solution would be to remove support for providing multiple features with the notation `(ft, fn1, fn2, fn3, ...)` so that this notation can be used with the single purpose of defining `(ft, fn, new_fn)`. I'll open a new internal issue for this.

Notation `(FeatureType.DATA: {'BANDS': 'NEW_BANDS'}}` is for sure much cleaner but in some cases a bit less convenient. Besides that, you probably need `new_names` parameter so that users could be informed if their new names will have an effect or not. The information if a features parameter accepts new names or not should probably be written in docstrings for each parameter. At the moment, this is in many cases a hidden/undocumented extra functionality for which you have to read the code to understand it.

Reply:
Completely agree about the new issue.

I would personally opt for `(FeatureType.DATA: {'BANDS': 'NEW_BANDS'}}` notation, which is familiar from how pandas rename works (`rename={'old_name': 'new_name'}). Although it seems less convenient, it would allow for "everything in one go":
`[(FeatureType.DATA: [{'old_name': 'new_name}, 'name_1', 'name_2', {'old_name3' : 'new_name3'}]`